packagingbuild/centos7/Dockerfile

```diff
@@ -77,14 +77,13 @@ RUN yum -y install nc net-tools
 #   - SSH access for build executor.
 #
 
-# Install development tools and python for EL8
-RUN yum -y install python2 \
-    python2-devel rpmdevtools
+# Install development tools and python2 for EL <=7
+RUN yum -y install python2 python2-devel rpmdevtools
 
 # Install fresh pip and certs
 RUN curl https://bootstrap.pypa.io/get-pip.py | \
     python - virtualenv==16.6.0 pip==19.1.1 wheel setuptools && \
-    pip install --upgrade requests[security] venvctrl && rm -rf /root/.cache
+RUN pip3 install requests[security] venvctrl --upgrade && rm -rf /root/.cache
```

Comment:
CentOS7 build failed: 
```
Installing collected packages: virtualenv, pip, wheel, setuptools
Successfully installed pip-19.1.1 setuptools-44.0.0 virtualenv-16.6.0 wheel-0.33.6
WARNING: You are using pip version 19.1.1; however, version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
/bin/sh: RUN: command not found
Removing intermediate container d0681a59f17b
The command '/bin/sh -c curl https://bootstrap.pypa.io/get-pip.py | python - virtualenv==16.6.0 pip==19.1.1 wheel setuptools && RUN pip3 install requests[security] venvctrl --upgrade && rm -rf /root/.cache' returned a non-zero code: 127
```
See previous line why.

Reply:
Also I don't like those minor syntax/line changes in every PR. This makes the diff harder to read and brings inconsistencies with other Dockerfiles, for example:
https://github.com/StackStorm/st2packaging-dockerfiles/blob/3b19d8489465d4d5dac5fcd0214f1c157b9318ee/packagingbuild/xenial/Dockerfile#L49-L51

Reply:
Oh - I understand - so the reason that this was split out is that pip isn't designed to be ran twice in one process. The problem I was having in EL8 was that since PIP wasn't exactly exiting, that it was not using the proper environment. This is not a problem in EL7 or EL6, but is a problem in EL8.

Since in the centos8/Dockerfile we needed to run one pip command that did an install, and a separate command that did an upgrade, I needed to make them separate run statements.